src/main/java/com/faforever/gw/model/ValidationHelper.java

```diff
@@ -55,7 +55,7 @@ public void validateAssaultOnPlanet(GwCharacter character, Planet planet) {
             throw gwErrorService.getBpmnErrorOf(GwErrorType.PLANET_PROTECTED_FROM_ASSAULT);
         }
 
-        if (false) {// TODO: Implement check whether character's faction can reach planet
+        if (!planet.getSolarSystem().isReachable(character.getFaction())) {
```

Comment:
This is not the only condition. The planet is also reachable if one of the other planets in the same solar systems is owned by the character's faction.